client/tasks/fills/PaymentGatewaySuggestions/index.js

```diff
@@ -165,27 +166,43 @@ export const PaymentGatewaySuggestions = ( { onComplete, query } ) => {
 
 	const [ wcPayGateway, enabledGateways, additionalGateways ] = useMemo(
 		() =>
-			Array.from( paymentGateways.values() ).reduce(
-				( all, gateway ) => {
-					const [ wcPay, enabled, additional ] = all;
-
-					// WCPay is handled separately when not installed and configured
-					if (
-						gateway.plugins?.length === 1 &&
-						gateway.plugins[ 0 ] === 'woocommerce-payments' &&
-						! ( gateway.installed && ! gateway.needsSetup )
-					) {
-						wcPay.push( gateway );
-					} else if ( gateway.enabled ) {
-						enabled.push( gateway );
-					} else {
-						additional.push( gateway );
+			Array.from( paymentGateways.values() )
+				.sort( ( a, b ) => {
+					if ( a.hasPlugins === b.hasPlugins ) {
+						return (
+							a.recommendation_priority -
+							b.recommendation_priority
+						);
```

Comment:
Nice!!
We do have the same sort logic just above this here: https://github.com/woocommerce/woocommerce-admin/blob/main/client/tasks/fills/PaymentGatewaySuggestions/index.js#L139-L149

It doesn't appear we can make use of `additionalGateways` there instead, so maybe we just want to move the sorting function into a helper function and use that in both of these sort methods. That way if we ever update the sort logic it will be updated for both instances.

Reply:
Sounds good. Changed in b6c27f5349f5738567745dd6f19a7c2de7137260.